ui-app/client/src/components/FormStorageEditor/components/formCommons.ts

```diff
@@ -0,0 +1,586 @@
+	[
+		'EMAIL',
+		{
+			uuid: '',
+			order: undefined,
+			type: 'EMAIL',
+			message: 'Not a valid email',
+		},
+	],
+	[
+		'PHONE',
+		{
+			uuid: '',
+			order: undefined,
+			type: 'REGEX',
+			pattern: '^(\\+\\d{1,3}\\s?)?\\(?\\d{3}?\\)?\\d{4,15}$',
+			message: 'Not a valid phone number',
+		},
+	],
+	[
+		'STRING_LENGTH',
+		{
+			uuid: '',
+			order: undefined,
+			type: 'STRING_LENGTH',
+		},
+	],
+	[
+		'NUMBER_VALUE',
+		{
+			uuid: '',
+			order: undefined,
+			type: 'NUMBER_VALUE',
+		},
+	],
+	[
+		'ALPHABETIC',
+		{
+			uuid: '',
+			order: undefined,
+			type: 'REGEX',
+			pattern: '^[a-zA-Z]+$',
+			message: '',
+		},
+	],
+	[
+		'ALPHANUMERIC',
+		{
+			uuid: '',
+			order: undefined,
+			type: 'REGEX',
+			pattern: '^[a-zA-Z0-9]+$',
+			message: '',
+		},
+	],
+	[
+		'URL',
+		{
+			uuid: '',
+			order: undefined,
+			type: 'REGEX',
+			pattern:
+				'\\b(?:https?|ftp):\\/\\/[-A-Za-z0-9+&@#\\/%?=~_|!:,.;]*[-A-Za-z0-9+&@#\\/%=~_|]',
+			message: '',
+		},
+	],
+	[
+		'CURRENCY',
+		{
+			uuid: '',
+			order: undefined,
+			type: 'REGEX',
+			pattern: '^₹?\\s?\\d{1,3}(?:,?\\d{3})*(?:\\.\\d{1,2})?$',
+			message: '',
+		},
+	],
+	[
+		'CHARACTERS_ONLY',
+		{
+			uuid: '',
+			order: undefined,
+			type: 'REGEX',
+			pattern: '^[^\\d]*$',
+			message: '',
+		},
+	],
+	[
+		'REGEX',
+		{
+			uuid: '',
+			order: undefined,
+			type: 'REGEX',
+			pattern: '',
+			message: '',
+		},
+	],
+]);
+
+export const previewCompDefinitionMap = new Map<string, ComponentDefinition>([
+	[
+		'TextBox',
+		{
+			key: '',
+			name: 'TextBox',
+			type: 'TextBox',
+			properties: {
+				showNumberSpinners: {
+					value: false,
+				},
+
+				placeholder: {
+					value: '',
+				},
+			},
+			displayOrder: undefined,
+			bindingPath: {
+				type: 'VALUE',
+				value: 'Page.textBox',
+			},
+		},
+	],
+	[
+		'TextArea',
+		{
+			key: '',
+			name: 'TextArea',
+			type: 'TextArea',
+			properties: {
+				placeholder: {
+					value: 'TextArea',
+				},
+			},
+			displayOrder: undefined,
+			bindingPath: {
+				type: 'VALUE',
+				value: 'Page.textArea',
+			},
+		},
+	],
+	[
+		'Dropdown',
+		{
+			key: '',
+			name: 'Dropdown',
+			type: 'Dropdown',
+			properties: {
+				placeholder: {
+					value: '',
+				},
+				selectionType: {
+					value: 'OBJECT',
+				},
+				uniqueKeyType: {
+					value: 'RANDOM',
+				},
+			},
+			displayOrder: undefined,
+			bindingPath: {
+				type: 'VALUE',
+				value: 'Page.selected',
+			},
+		},
+	],
+	[
+		'RadioButton',
+		{
+			key: '2o2Bqo4cXW2Zxa0q5J8EdQ',
+			name: 'RadioButton',
+			type: 'RadioButton',
+			properties: {
+				selectionType: {
+					value: 'OBJECT',
+				},
+
+				layout: {
+					value: 'ROWLAYOUT',
+				},
+				uniqueKeyType: {
+					value: 'OBJECT',
+				},
+			},
+			displayOrder: undefined,
+			bindingPath: {
+				type: 'VALUE',
+				value: 'Page.radioSelected',
+			},
+		},
+	],
+]);
+
+export interface PreviewCompValidationProperties {
+	type: string;
+	message?: { value?: string };
```

Comment:
If there is no value then there won't be a message right? So, you don't make the value optional, just it's parent

`message?: {value: string};`